lib/active_admin/filters/humanized.rb

```diff
@@ -9,8 +9,22 @@ def initialize(param)
         @value = param[1]
       end
 
+      def related_class
+        @related_class ||=
+            begin
+              @body.split('_id').first.gsub(' ', '').classify.constantize
+            rescue
+              nil
+            end
+      end
+
       def value
-        @value
+        @value =
+            begin
+              display_name(related_class.find(@value))
```

Comment:
@Shoroh You should be able to do:

`related_class.find(@value).display_name`

If they don't have it defined, it'll just default to `@value` below.